Java/core/src/main/java/com/amazon/randomcutforest/store/PointStore.java

```diff
@@ -28,44 +28,224 @@
  * point values and increment and decrement reference counts. Valid index values
  * are between 0 (inclusive) and capacity (exclusive).
  */
-public abstract class PointStore<Point> extends IndexManager implements IPointStore<Point> {
+public abstract class PointStore<Store, P> extends IndexManager implements IPointStore<P> {
```

Comment:
Why did we need to add this generic parameter? In the only existing subclasses, the types of Store and P are the same. What's the motivation to make these separate type parameters/

Reply:
Currently we are using double[] or float[] as examples of the generic Point. We were using double[] and float[] for the store class with duplication of code. That duplication is avoided if Store is parametrized as well, that's the proximate reason. Otherwise this could be double[][], or P[] which would lose space efficiency.